packages/firestore/src/core/firestore_client.ts

```diff
@@ -486,24 +480,17 @@ export class FirestoreClient {
     return this.databaseInfo.databaseId;
   }
 
-  addSnapshotsInSyncListener(observer: Observer<void>): void {
+  addSnapshotsInSyncListener(observer: AsyncObserver<void>): Unsubscribe {
     this.verifyNotTerminated();
-    this.asyncQueue.enqueueAndForget(() => {
-      this.eventMgr.addSnapshotsInSyncListener(observer);
-      return Promise.resolve();
-    });
-  }
-
-  removeSnapshotsInSyncListener(observer: Observer<void>): void {
-    // Checks for shutdown but does not raise error, allowing remove after
-    // shutdown to be a no-op.
-    if (this.clientTerminated) {
-      return;
-    }
-    this.asyncQueue.enqueueAndForget(() => {
-      this.eventMgr.removeSnapshotsInSyncListener(observer);
-      return Promise.resolve();
-    });
+    this.asyncQueue.enqueueAndForget(async () =>
```

Comment:
Why is it necessary to specify an async function here to `this.asyncQueue.enqueueAndForget()`? In the `listen()` method it just specifies a normal, non-async function. I see that this is consistent with the old code but the reason for the inconsistency within this file is non-obvious to me. Should all calls to `enqueueAndForget()` be async functions? It would appear so based on the signature of `enqueueAndForget()`.

Reply:
There is a test already: https://github.com/firebase/firebase-js-sdk/blob/ea699fac1220f6d1de3f719c00d96f5039e388ed/packages/firestore/test/integration/api/database.test.ts#L1235

Reply:
Ahh perfect.

Reply:
Sorry - I replied to the wrong comment (but looks like you managed to restore the context). `async` makes the function return a Promise, even if the return of the value being returned is not a Promise. In the case above, "eventManager.listen" already returns a Promise.

This change is not needed, it just felt right to me. Manually returning `Promise.resolve` works as well.

Reply:
That makes sense. I forgot to check the return type of `eventManager.listen()`. With this explanation, I'd recommend to keep it as is, with the more modern `async` keyword.